app/main.py

```diff
@@ -1,2 +1,81 @@
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.threshold = 2 / 3
+        self.length = 0
+        self.hash_table: list = [None] * self.capacity
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        cell_index = self.calculate_index(key)
+
+        if self.hash_table[cell_index] is None:
+            self.hash_table[cell_index] = (key, value)
+            self.length += 1
```

Comment:
Your node should contain a hash key as third attribute. Something like this `(key, hash_key, value)`